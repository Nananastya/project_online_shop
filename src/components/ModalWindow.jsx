import React from 'react';
import PropTypes from "prop-types";
import Modal from 'react-bootstrap/Modal';
import "./../scss/style.scss";

export default function ModalWindow(props) {
        let headerBg, bodyBg;
        if (props.color === "red"){
            headerBg = "#b30000";
            bodyBg = "#ea2e2e";
        }
        else if (props.color === "blue"){
            headerBg = "#03406A";
            bodyBg = "#3E94D1";
        }
        
        return (
            <Modal className='modal-window' show={props.show1} onHide={props.handleClose}>
                <Modal.Header className='modal-header' style={{backgroundColor: `${headerBg}`}} closeButton={props.closeButton} closeVariant='white'>
                    <Modal.Title>{props.header}</Modal.Title>
                </Modal.Header>
                <Modal.Body className='modal-body ' style={{width:"100%", backgroundColor: `${bodyBg}`}}>
                    {props.text}
                    <p>{ 
                            props.lastChosenProduct &&
                            <img 
                                    style={{
                                        marginTop: "10px",
                                        height: '100px',
                                        width: '110px',
                                    }}
                                src={props.lastChosenProduct.url} alt="food" />
                        }</p>
                    </Modal.Body>
                    <Modal.Footer className='modal-footer' style={{backgroundColor: `${bodyBg}`}}>
                        {props.actions}
                    </Modal.Footer>
            </Modal>
        )
}


ModalWindow.propTypes = {
    show1: PropTypes.bool,
    text: PropTypes.node,
    header: PropTypes.string,
    lastChosenProduct: PropTypes.object,
    actions: PropTypes.element,
    handleClose: PropTypes.func,
    closeButton: PropTypes.bool
}