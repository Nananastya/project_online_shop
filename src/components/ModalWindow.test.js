import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import ModalWindow from "./ModalWindow";

describe("ModalWindow component", () => {
  const handleClose = jest.fn();
  const lastChosenProduct = {
    name: "apple",
    price: 100,
    url: "https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fstatic.onecms.io%2Fwp-content%2Fuploads%2Fsites%2F39%2F2012%2F06%2F20222932%2Frome_3.jpg",
    articul: 15339,
    color: "light-red"
  };
  const header = "Are you sure?";
  const text = "Are you sure you want to delete this product from the cart?";
  const actions = (
    <>
      <button data-testid="ok-button" onClick={handleClose}>
        Ok
      </button>
      <button data-testid="no-button" onClick={handleClose}>
        No
      </button>
    </>
  );
  const props = {
    show1: true,
    handleClose: handleClose,
    closeButton: true,
    header: header,
    text: text,
    lastChosenProduct: lastChosenProduct,
    actions: actions,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render the modal with correct header and text", () => {
    const { container } = render(<ModalWindow {...props} />);
    const headerElement = screen.getByText(header);
    const textElement = screen.getByText(text);
    expect(headerElement).toBeDefined();
    expect(textElement).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it("should render the modal with an image of last chosen product if it exists", () => {
    render(<ModalWindow {...props} />);
    const imageElement = screen.getByAltText("food");
    expect(imageElement).toBeDefined();
  });

  it("should call handleClose function when Ok or No button is clicked", () => {
    render(<ModalWindow {...props} />);
    const okButton = screen.getByTestId("ok-button");
    const noButton = screen.getByTestId("no-button");
    fireEvent.click(okButton);
    fireEvent.click(noButton);
    expect(handleClose).toHaveBeenCalledTimes(2);
  });
});

