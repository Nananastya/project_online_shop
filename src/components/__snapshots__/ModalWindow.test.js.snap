// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWindow component should render the modal with correct header and text 1`] = `<div />`;
